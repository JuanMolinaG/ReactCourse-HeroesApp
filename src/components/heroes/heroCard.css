.card-container {
  display: grid;
  perspective: 700px;
}

.card-flip {
  display: grid;
  grid-template: 1fr / 1fr;
  grid-template-areas: 'frontAndBack';
  transform-style: preserve-3d;
  transition: all 0.7s ease;
}

.card-flip div {
  backface-visibility: hidden;
  transform-style: preserve-3d;
}

.front {
  grid-area: frontAndBack;
}

.front img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.back {
  grid-area: frontAndBack;
  transform: rotateY(-180deg);
}

.card-container:hover .card-flip {
  transform: rotateY(180deg);
}
