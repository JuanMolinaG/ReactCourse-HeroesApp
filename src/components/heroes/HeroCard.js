import React from 'react';
import { Link } from 'react-router-dom';
import './heroCard.css';

export const HeroCard = ({
  id,
  superhero,
  publisher,
  alter_ego,
  first_appearance,
  characters,
}) => {
  return (
    <div className='card-container col-md-4 col-sm-6'>
      <div className='card-flip'>
        <div className='card front'>
          <img
            src={`./assets/heroes/${id}.jpg`}
            className='card-img'
            alt={superhero}
          />
        </div>
        <div className='card back p-4'>
          <div className='card-block'>
            <h5 className='card-title'>{superhero}</h5>
            <p className='card-text'>
              <b>Alter Ego:</b> {alter_ego}
            </p>
            {alter_ego !== characters && (
              <p className='card-text'>
                <b>Characters:</b> {characters}
              </p>
            )}
            <p className='card-text'>
              <small className='text-muted'>
                <b>First Appeatance:</b> {first_appearance}
              </small>
            </p>
            <Link to={`./hero/${id}`}>More...</Link>
          </div>
        </div>
      </div>
    </div>
  );
};
