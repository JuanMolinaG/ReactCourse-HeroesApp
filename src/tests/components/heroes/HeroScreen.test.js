import React from 'react';
import { mount } from 'enzyme';
import { HeroScreen } from '../../../components/heroes/HeroScreen';
import { MemoryRouter, Route } from 'react-router-dom';

describe('Pruebas en <HeroScreen />', () => {
  const historyMock = {
    lenght: 10,
    push: jest.fn(),
    goBack: jest.fn(),
  };

  test('Debe de mostrar el componente Redirect si no hay argumentos en el URL', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/hero']}>
        <HeroScreen history={historyMock} />
      </MemoryRouter>
    );

    expect(wrapper.find('Redirect').exists()).toBe(true);
  });

  test('Debe de mostrar un hero si el parÃ¡metro existe y se encuentra el hero', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/hero/dc-batman']}>
        <Route path='/hero/:heroId' component={HeroScreen} />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.row').exists()).toBe(true);
  });

  test('Debe de regresar a la pantalla principal del hero con push', () => {
    const historyMock = {
      length: 1,
      push: jest.fn(),
      goBack: jest.fn(),
    };

    const wrapper = mount(
      <MemoryRouter initialEntries={['/hero/dc-batman']}>
        <Route
          path='/hero/:heroId'
          component={() => <HeroScreen history={historyMock} />}
        />
      </MemoryRouter>
    );

    wrapper.find('button').prop('onClick')();

    expect(historyMock.push).toHaveBeenCalledWith('/dc');
    expect(historyMock.goBack).not.toHaveBeenCalled();
  });

  test('Debe de regresar a la pantalla anterior con goBack', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/hero/dc-batman']}>
        <Route
          path='/hero/:heroId'
          component={() => <HeroScreen history={historyMock} />}
        />
      </MemoryRouter>
    );

    wrapper.find('button').prop('onClick')();

    expect(historyMock.goBack).toHaveBeenCalled();
    expect(historyMock.push).toHaveBeenCalledTimes(0);
  });

  test('Debe de llamar el redirect si el hero no existe', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/hero/dc-batsy']}>
        <Route
          path='/hero/:heroId'
          component={() => <HeroScreen history={historyMock} />}
        />
      </MemoryRouter>
    );

    expect(wrapper.text()).toBe('');
  });
});
