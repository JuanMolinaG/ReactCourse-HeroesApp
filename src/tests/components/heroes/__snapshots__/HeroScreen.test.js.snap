// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroScreen /> Debe de mostrar un hero si el par√°metro existe y se encuentra el hero 1`] = `
<div
  className="row mt-5"
>
  <div
    className="col-4"
  >
    <img
      alt="Batman"
      className="img-thumbnail animate__animated animate__fadeInLeft"
      src="../assets/heroes/dc-batman.jpg"
    />
  </div>
  <div
    className="col-8"
  >
    <h3>
      Batman
    </h3>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        <b>
          Alter Ego: 
        </b>
        Bruce Wayne
      </li>
      <li
        className="list-group-item"
      >
        <b>
          Publisher: 
        </b>
        DC Comics
      </li>
      <li
        className="list-group-item"
      >
        <b>
          First appearance: 
        </b>
        Detective Comics #27
      </li>
    </ul>
    <h5>
      Characters
    </h5>
    <p>
      Bruce Wayne
    </p>
    <button
      className="btn btn-outline-info"
      onClick={[Function]}
    >
      Return
    </button>
  </div>
</div>
`;
